src/main/java/com/cb/controllers/LoginController.java

```diff
@@ -34,31 +33,28 @@
 
     @RequestMapping(value = "/loginuser", method = RequestMethod.POST)
     public String loginUser(HttpServletRequest req, Map<String, String> model, @ModelAttribute("userDAL") UserBL userBL, Model p, Model c) {
-        List<UserDAL> allEmails = userService.getUserByEmail(userBL.getEmail());
-        List<UserDAL> allPasswords = userService.getUserByPassword(userBL.getPassword());
-        String userName = "";
-        for (UserDAL items: allEmails
-        ) {
-            userName = items.getUserName();
-        }
+        int userExist = userService.getUserByEmailAndPassword(userBL);
 
-        if (allEmails.size() > 0 && allPasswords.size() > 0) {
+        if (userExist == 1) {
+            String userName = userService.getUserNameByEmail(userBL);
             HttpSession userSession = req.getSession();
             userSession.setAttribute("userName", userName);
             List<PartyDAL> partiesList = partyService.getParties();
             List<CharacterDAL> charactersList = characterService.getCharacters();
             p.addAttribute("partiesList", partiesList);
             c.addAttribute("charactersList", charactersList);
             return "createCharacter";
-
-        } else {
+        } else if (userExist == -1) {
             model.put("error", "User does not exist");
             return "index";
-
+        } else {
+            model.put("error", "General error");
+            return "index";
```

Comment:
Please comment what is general error?

Reply:
I've changed it. Now it returns the errorPage. Type of exception we could not handle